contracts/ReversibleICO.sol

```diff
@@ -242,35 +244,35 @@ contract ReversibleICO is IERC777Recipient {
         Stage storage stage0 = stages[stageCount];
         // stageCount = 0
         stage0.startBlock = _commitPhaseStartBlock;
-        stage0.endBlock = _commitPhaseStartBlock + _commitPhaseBlockCount;
+        stage0.endBlock = commitPhaseEndBlock;
         stage0.tokenPrice = _commitPhasePrice;
 
         stageCount++;
         // stageCount = 1
 
 
         // Setup stage 1 to n: The buy phase stages
-        uint256 lastStageBlockEnd = stage0.endBlock;
+        // Each new stage starts after the previous phase's endBlock
+        uint256 newStageEndBlock = stage0.endBlock;
 
         for (uint8 i = 1; i <= _stageCount; i++) {
 
             Stage storage stageN = stages[stageCount];
             // stageCount = n
-            // Each new stage starts after the previous phase's endBlock
-            stageN.startBlock = lastStageBlockEnd + 1;
-            stageN.endBlock = lastStageBlockEnd + _stageBlockCount + 1;
+            stageN.startBlock = newStageEndBlock + 1;
+            stageN.endBlock = newStageEndBlock + _stageBlockCount;
             // At each stage the token price increases by _stagePriceIncrease * stageCount
             stageN.tokenPrice = _commitPhasePrice + (_stagePriceIncrease * (i));
             stageCount++;
 
-            lastStageBlockEnd = stageN.endBlock;
+            newStageEndBlock = stageN.endBlock;
         }
 
         // The buy phase starts on the subsequent block of the commitPhase's (stage0) endBlock
         buyPhaseStartBlock = commitPhaseEndBlock + 1;
-        buyPhaseEndBlock = lastStageBlockEnd;
+        buyPhaseEndBlock = newStageEndBlock;
         // The duration of buyPhase in blocks
-        buyPhaseBlockCount = lastStageBlockEnd - buyPhaseStartBlock;
+        buyPhaseBlockCount = newStageEndBlock - buyPhaseStartBlock + 1;
```

Comment:
Should convert operations to safeMath, maybe not now but after tests are updated to work with current changes.
i.e.
```
        buyPhaseBlockCount = newStageEndBlock.sub( buyPhaseStartBlock ).add(1); 
```

Reply:
True, I  just wanted to create a separate PR for this.